conu/backend/docker/image.py

```diff
@@ -232,6 +233,31 @@ def callback():
 
         return DockerContainer(self, container_id, popen_instance=popen_instance, name=container_name)
 
+    def has_pkgs_signed_with(self, allowed_keys):
+        """
+        Check signature of packages installed in image.
+        Raises exception when
+         * rpm binary is not installed in image
+         * parsing of rpm fails
+         * there are packages in image that are not signed with one of allowed keys
+
+        :param allowed_keys: list of allowed keys
+        :return: bool
+        """
+
+        if not allowed_keys or not isinstance(allowed_keys, list):
+            raise ConuException("allowed_keys must be a list")
+        self.pull()
+        drb = DockerRunBuilder(command=['rpm', '-qa', '--qf', '%{name} %{SIGPGP:pgpsig}\n'])
+        cont = self.run_via_binary(drb)
+        out = cont.logs_unicode()[:-1].split('\n')
```

Comment:
even logs can actually throw an exc so I would put it inside the try/finally block